theano/tensor/tests/test_nlinalg.py

```diff
@@ -177,6 +178,34 @@ def test_tensorsolve():
     t_x = fn(a, b)
     assert _allclose(n_x, t_x)
 
+    # check the type upcast now
+    C = tensor.tensor4("C", dtype='float32')
+    D = tensor.matrix("D", dtype='float64')
+    Y = tensorsolve(C, D)
+    fn = function([C, D], [Y])
+
+    c = numpy.eye(2*3*4).astype('float32')
```

Comment:
nitpik, this can be done like "numpy.eye(2*3*4, dtype='float32')". It is a little bit faster and more readable.